xarray/core/datatree.py

```diff
@@ -1896,3 +1901,181 @@ def apply_indexers(dataset, node_indexers):
+        chunks.
+
+        Non-dask arrays in this tree will be converted to dask arrays. Dask
+        arrays will be rechunked to the given chunk sizes.
+
+        If neither chunks is not provided for one or more dimensions, chunk
+        sizes along that dimension will not be updated; non-dask arrays will be
+        converted into dask arrays with a single block.
+
+        Along datetime-like dimensions, a :py:class:`groupers.TimeResampler` object is also accepted.
+
+        Parameters
+        ----------
+        chunks : int, tuple of int, "auto" or mapping of hashable to int or a TimeResampler, optional
+            Chunk sizes along each dimension, e.g., ``5``, ``"auto"``, or
+            ``{"x": 5, "y": 5}`` or ``{"x": 5, "time": TimeResampler(freq="YE")}``.
+        name_prefix : str, default: "xarray-"
+            Prefix for the name of any new dask arrays.
+        token : str, optional
+            Token uniquely identifying this dataset.
+        lock : bool, default: False
+            Passed on to :py:func:`dask.array.from_array`, if the array is not
+            already as dask array.
+        inline_array: bool, default: False
+            Passed on to :py:func:`dask.array.from_array`, if the array is not
+            already as dask array.
+        chunked_array_type: str, optional
+            Which chunked array type to coerce this datasets' arrays to.
+            Defaults to 'dask' if installed, else whatever is registered via the `ChunkManagerEntryPoint` system.
+            Experimental API that should not be relied upon.
+        from_array_kwargs: dict, optional
+            Additional keyword arguments passed on to the `ChunkManagerEntrypoint.from_array` method used to create
+            chunked arrays, via whichever chunk manager is specified through the `chunked_array_type` kwarg.
+            For example, with dask as the default chunked array type, this method would pass additional kwargs
+            to :py:func:`dask.array.from_array`. Experimental API that should not be relied upon.
+        **chunks_kwargs : {dim: chunks, ...}, optional
+            The keyword arguments form of ``chunks``.
+            One of chunks or chunks_kwargs must be provided
+
+        Returns
+        -------
+        chunked : xarray.DataTree
+
+        See Also
+        --------
+        Dataset.chunk
+        Dataset.chunksizes
+        xarray.unify_chunks
+        dask.array.from_array
+        """
+        return DataTree.from_dict(
+            {
+                path: node.dataset.chunk(
+                    chunks,
+                    name_prefix=name_prefix,
+                    token=token,
+                    lock=lock,
+                    inline_array=inline_array,
+                    chunked_array_type=chunked_array_type,
+                    from_array_kwargs=from_array_kwargs,
+                    **chunks_kwargs,
+                )
```

Comment:
this will apply `chunks` / `chunk_kwargs` to all nodes equally, even if some of the nodes don't have all dimensions... should we filter the chunk sizes by the dims on each node?

Reply:
We probably should... I think this is equivalent to the `_selective_indexing` solution in https://github.com/pydata/xarray/pull/9588/files.

Reply:
Do you want me to pick this PR up and add that part?

Reply:
as you wish, but now that I'm back home I can also look into it

Reply:
Whatever will be faster! I'm itching to release datatree 😅 Just let me know if you want me to take over.

Reply:
In the end I'm doing something a bit less complicated than `_selective_indexing`, but this allows passing dims that don't appear in the `DataTree`. Should we guard against that?

Reply:
> In the end I'm doing something a bit less complicated than `_selective_indexing`

Your implementation looks fine too!

> Should we guard against that?

I just checked and `Dataset` raises if you try that so `DataTree` ideally should too.

It should be relatively easy to implement by checking against [`DataTree._get_all_dims`](https://github.com/pydata/xarray/blob/521b087e96f637606daec7250f7953440e5f9707/xarray/core/datatree.py#L1692) up front?